File: examples/typescript/memory.ts
/**
 * Memory Example - TypeScript
 *
 * This example shows how to use Nexi's memory system.
 * Run with: npx tsx examples/typescript/memory.ts
 */

import { NexiAPI } from '../../src/api/index.js';

async function main() {
  console.log('🤖 Nexi AI - Memory Example\n');

  const nexi = await NexiAPI.create({
    dataDir: './example-data-memory',
    logLevel: 'none',
  });

  // Store different types of memories
  console.log('📝 Storing memories...\n');

  nexi.remember('User prefers dark mode', { type: 'preference', importance: 6 });
  nexi.remember('User works as a software developer', { type: 'fact', importance: 8 });
  nexi.remember('User mentioned they have a cat named Luna', { type: 'fact', importance: 7 });
  nexi.remember('Had a great conversation about TypeScript', { type: 'event', importance: 5 });

  // Search memories
  console.log('🔍 Searching for "developer":');
  const results = nexi.searchMemories('developer');
  for (const memory of results.memories) {
    console.log(`  - [${memory.type}] ${memory.content} (importance: ${memory.importance})`);
  }
  console.log();

  // Get stats
  const stats = nexi.getStats();
  console.log('📊 Memory stats:');
  console.log(`  Total memories: ${stats.memories.total}`);
  console.log(`  By type:`, stats.memories.byType);
  console.log(`  Average importance: ${stats.memories.averageImportance.toFixed(1)}`);
  console.log();

  // Chat - memories will be used in context
  if (await nexi.isAvailable()) {
    console.log('💬 Chat (memories will influence response):');
    console.log('You: Do you remember anything about me?');
    const response = await nexi.chat('Do you remember anything about me?');
    console.log(`Nexi: ${response.message}`);
  }

  nexi.shutdown();
  console.log('\n✨ Done!');
}

main().catch(console.error);

File: examples/python/memory.py
"""
Memory Example - Python

This example shows how to use Nexi's memory system.
Run with: python examples/python/memory.py
"""

import sys
sys.path.insert(0, './python')

from nexi import Nexi, NexiConfig, MemoryType

def main():
    print("🤖 Nexi AI - Memory Example\n")

    config = NexiConfig(data_dir="./example-data-py-memory")
    nexi = Nexi(config)

    # Store different types of memories
    print("📝 Storing memories...\n")

    nexi.remember("User prefers dark mode", memory_type=MemoryType.PREFERENCE, importance=6)
    nexi.remember("User works as a software developer", memory_type=MemoryType.FACT, importance=8)
    nexi.remember("User mentioned they have a cat named Luna", memory_type=MemoryType.FACT, importance=7)
    nexi.remember("Had a great conversation about Python", memory_type=MemoryType.EVENT, importance=5)

    # Search memories
    print('🔍 Searching for "developer":')
    results = nexi.search_memories("developer")
    for memory in results.memories:
        print(f"  - [{memory.type.value}] {memory.content} (importance: {memory.importance})")
    print()

    # Get all memories
    all_memories = nexi.get_all_memories()
    print(f"📚 All memories ({len(all_memories)}):")
    for memory in all_memories:
        print(f"  - [{memory.type.value}] {memory.content}")
    print()

    # Get stats
    stats = nexi.get_stats()
    print("📊 Memory stats:")
    print(f"  Total memories: {stats.total_memories}")
    print(f"  By type: {stats.memories_by_type}")
    print(f"  Average importance: {stats.average_importance:.1f}")
    print()

    # Chat - memories will be used in context
    if nexi.is_available():
        print("💬 Chat (memories will influence response):")
        print("You: Do you remember anything about me?")
        response = nexi.chat("Do you remember anything about me?")
        print(f"Nexi: {response.message}")

    nexi.shutdown()
    print("\n✨ Done!")


if __name__ == "__main__":
    main()
